### PR TITLE
Checkstyle plugin with google style

### PR DESCRIPTION
**Pull request by PepperJo**
Add checkstyle plugin but do not check on compilation yet.
Checkstyle can be run with 'mvn checkstyle:check'

https://issues.apache.org/jira/projects/CRAIL/issues/CRAIL-19

Signed-off-by: Jonas Pfefferle <pepperjo@apache.org>

### CODE REVIEW

**Comment by animeshtrivedi** [comment]
Lets merge this. Also I think, having something like this would be useful for DaRPC and DiSNI as well?

**Comment by patrickstuedi** [comment]
Why not add the checkstyle plugin to the top level pom?

**Comment by PepperJo** [comment]
Let me try again last time I checked it did not work as expected.

**Comment by PepperJo** [comment]
Moved plugin to parent pom. Note that I do not yet enforce style check on compilation but you can manually run checkstyle with 'mvn checkstyle:check'. I propose whenever someone changes a file to make two commits one that fixes the style and one for the issue. Otherwise solving all the errors would probably take a few hours...

**Comment by PepperJo** [comment]
Also if someone is not happy with the suggestions of the style checker please make a pull request to change the google style derived xml file.

**Comment by patrickstuedi** [comment]
Looks good. Can we just rename google_check.xml to checkstyle.xml?

**Comment by PepperJo** [comment]
sure

**Comment by PepperJo** [comment]
Renamed file.
